prov/sockets/src/sock_ep_msg.c

```diff
@@ -1235,3 +1341,154 @@ int sock_msg_passive_ep(struct fid_fabric *fabric, struct fi_info *info,
 	free(_pep);
 	return ret;
 }
+
+static void sock_ep_cm_handle_rx(struct sock_ep_cm_head *cm_head,
+                                struct sock_conn_req_handle *handle)
+{
+	struct sock_conn_hdr hdr;
+
+	if(sock_cm_recv(handle->sock_fd, &hdr, sizeof(hdr))) {
+		SOCK_LOG_ERROR("io failed for fd %d\n", handle->sock_fd);
+		if (handle->ep) {
+			if (!handle->pep) /* Active-side */
+				sock_ep_cm_report_connect_fail(handle->ep, NULL, 0);
+			sock_ep_cm_shutdown_report(handle->ep, 0);
```

Comment:
Reporting a connection failure and shutdown event for the same connection request looks wrong.

Reply:
Done. I rewrote this part accordingly.